### PR TITLE
Hotfix/remove use datasets

### PR DESCRIPTION
**Pull request by gokulavasan**
Remove/deprecate the use of useDatasets method in MR and Workers

JIRA: https://issues.cask.co/browse/CDAP-1845

Bamboo : http://builds.cask.co/browse/CDAP-DUT1116

### CODE REVIEW

File: cdap-api/src/main/java/co/cask/cdap/api/mapreduce/AbstractMapReduce.java [hunk 1]
```diff
@@ -91,7 +91,9 @@ protected final void setReducerResources(Resources resources) {
 
   /**
    * Specifies set of Dataset names that are used by the {@link MapReduce}.
+   * @deprecated datasets used in runtime need not be specified during configuration
```
Comment (ParthGandhi) [inline cdap-api/src/main/java/co/cask/cdap/api/mapreduce/AbstractMapReduce.java]: `... specified during MapReduce#configure()` i.e. link to method.

File: cdap-api/src/main/java/co/cask/cdap/api/mapreduce/AbstractMapReduce.java [hunk 2]
```diff
@@ -91,7 +91,9 @@ protected final void setReducerResources(Resources resources) {
...
    */
+  @Deprecated
   protected final void useDatasets(String dataset, String...moreDatasets) {
     List<String> datasets = new ArrayList<String>();
     datasets.add(dataset);
```
Comment (ParthGandhi) [inline cdap-api/src/main/java/co/cask/cdap/api/mapreduce/AbstractMapReduce.java]: Should we make this operation a no-op then? Why maintain a list?

File: cdap-api/src/main/java/co/cask/cdap/api/mapreduce/AbstractMapReduce.java [hunk 3]
```diff
@@ -101,7 +103,9 @@ protected final void useDatasets(String dataset, String...moreDatasets) {
 
   /**
    * Specifies set of Dataset names that are used by the {@link MapReduce}.
+   * @deprecated datasets used in runtime need not be specified during configuration
```
Comment (ParthGandhi) [inline cdap-api/src/main/java/co/cask/cdap/api/mapreduce/AbstractMapReduce.java]: Ditto.

File: cdap-api/src/main/java/co/cask/cdap/api/worker/WorkerConfigurer.java [hunk 4]
```diff
@@ -58,6 +58,8 @@
   /**
    * Adds the names of {@link Dataset Datasets} used by the worker.
    * @param datasets dataset names
+   * @deprecated datasets being used during runtime need not be specified during configuration
    */
+  @Deprecated
```
Comment (ParthGandhi) [inline cdap-api/src/main/java/co/cask/cdap/api/worker/WorkerConfigurer.java]: Ditto for above 2 comments.

**Comment by ParthGandhi** [comment]
A couple of comments. Rest LGTM.

File: cdap-api/src/main/java/co/cask/cdap/api/mapreduce/AbstractMapReduce.java (hunk 2 above)
Comment (gokulavasan) [inline cdap-api/src/main/java/co/cask/cdap/api/mapreduce/AbstractMapReduce.java]: We have marked it as deprecated this release. So if the user is still using useDatasets, he can get the info about MR, Workers associated with the dataset. We can remove this logic and the method completely in 3.0.

**Comment by gokulavasan** [comment]
@ParthGandhi Addressed your comments. Please take a look. Thanks!

File: cdap-api/src/main/java/co/cask/cdap/api/mapreduce/AbstractMapReduce.java (hunk 2 above)
Comment (ParthGandhi) [inline cdap-api/src/main/java/co/cask/cdap/api/mapreduce/AbstractMapReduce.java]: Yes, makes sense.

File: cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppFabricDataHttpHandler.java [hunk 5]
```diff
@@ -160,4 +148,16 @@ public void getMapReduceByDataset(HttpRequest request, HttpResponder responder,
     programListByDataAccess(responder, store, dsFramework, ProgramType.MAPREDUCE, Data.DATASET,
                             namespaceId, datasetId);
   }
+
```
Comment (ParthGandhi) [inline cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppFabricDataHttpHandler.java]: Can you move this back to where it used to be? It avoids an unnecessary diff.

**Comment by ParthGandhi** [comment]
Just one comment. The rest looks good.

**Comment by gokulavasan** [comment]
@ParthGandhi  Addressed your comment. Merging.
